### PR TITLE
refactor: enrich agent CLI schema output and simplify introspection (#418)

### PR DESCRIPTION
**Pull request by johnnygreco**
## Summary

Agent benchmark sessions showed that every session needing judge or code columns fell back to Python introspection (`help(Score)`, `print(list(CodeLang))`) because the schema output lacked nested type details and enum values — costing 2-3 extra tool calls per session.

This PR enriches the agent CLI schema output so agents get everything they need in a single `agent schema` call, while simplifying the underlying implementation.

### Architecture

Three focused modules in `cli/utils/`:
- **`agent_schema_view.py`** (new) — Pydantic model → structured field views (`PydanticModelView` / `PydanticFieldView` dataclasses). Uses `FieldInfo.annotation` directly (Pydantic v2 resolves annotations at model creation time, so no MRO walk needed).
- **`agent_introspection.py`** — Type discovery, catalog assembly, hidden field policy. Pure functions (no service class).
- **`agent_text_formatter.py`** — Structured views → human-readable text output.

### Changes

**Schema view layer** (`agent_schema_view.py`)
- Describe Pydantic models as structured metadata for downstream renderers
- Show enum member values inline (e.g., `values: bash, c, python, ...`)
- Expand nested `BaseModel` fields 1 level deep (e.g., `Score` inside `LLMJudgeColumnConfig`)
- Deduplicate nested model expansion (first field wins)
- Handle `Annotated` wrappers in nested generic args (e.g., `dict[str, Annotated[...]]`)

**Text formatter** (`agent_text_formatter.py`)
- Append instantiation example from required fields (e.g., `Example: dd.LLMCodeColumnConfig(name=..., ...)`)
- Render nested models indented under their parent field

**Introspection simplification** (`agent_introspection.py`)
- Removed `AgentContextService` class — all functions are stateless, so free functions are simpler
- Moved from `cli/services/` to `cli/utils/` (services in this codebase are stateful with injected dependencies)
- Declarative hidden field policy: `_FAMILY_HIDDEN_FIELDS` + `_TYPE_VISIBLE_OVERRIDES` dicts control which fields are hidden per family/type, keeping config models clean
- `allow_resize` hidden for all column types except `custom`
- Eliminated redundant `get_family_spec` lookups via internal `_discover_types(spec)` helper
- Shared `strip_module_paths` utility between schema view and signature formatting

**Config layer cleanup** (`base.py`)
- Removed `ConfigBase.schema_text()` and all supporting helpers (`_format_annotation`, `_get_docstring_summary`, `_GOOGLE_SECTION_HEADERS`) — schema rendering now lives entirely in the CLI layer
- Added `Field(description=...)` to `SingleColumnConfig.drop` and `SingleColumnConfig.allow_resize`

**Supporting changes**
- Added docstrings to `ScalarInequalityConstraint` and `ColumnInequalityConstraint`
- Updated `DropColumnsProcessorConfig` docstring
- Suppress PyArrow warnings inside `main()` only

## Test plan

- 31 unit tests for schema view covering all field variants, discriminator suppression, enum values, nested expansion (direct, list, optional, inherited), depth limiting, deduplication, union non-expansion, hidden fields, all-fields-hidden edge case, docstring truncation
- 11 introspection tests covering family discovery, catalog sorting, schema payloads, builder API, operations
- 4 integration tests for model aliases state, persona datasets, and full context assembly
- 17 formatter tests including 5 parametrized real-config integration tests (llm-text, sampler, category, code, scalar_inequality) plus judge schema expansion and custom column allow_resize visibility

### CODE REVIEW

File: packages/data-designer/src/data_designer/cli/main.py [hunk 1]
```diff
@@ -3,8 +3,13 @@
 
 from __future__ import annotations
 
+import warnings
+
 import typer
 
+warnings.filterwarnings("ignore", category=DeprecationWarning, module="pyarrow")
+warnings.filterwarnings("ignore", category=FutureWarning, module="pyarrow")
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer/src/data_designer/cli/main.py]: Good catch — fixed in 81caea01. Using `r"pyarrow(?:\\..*)?")` to match the top-level module and all submodules.

File: packages/data-designer-config/src/data_designer/config/base_utils.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+
+def _get_docstring_summary(docstring: str | None) -> str | None:
+    """Extract the first paragraph of a docstring, before any Google-style section header."""
+    if not docstring:
+        return None
+    lines: list[str] = []
+    for line in docstring.strip().splitlines():
+        stripped = line.strip()
+        if stripped.lower() in _GOOGLE_SECTION_HEADERS:
+            break
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer-config/src/data_designer/config/base_utils.py]: Stale — this referenced the old `get_type_hints()` fallback. The current code uses per-field `inspect.get_annotations()` with a targeted `NameError` catch.

File: packages/data-designer-config/src/data_designer/config/base_utils.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    required_field_names: list[str] = []
+    for name, field_info in cls.model_fields.items():
+        if _is_discriminator_field(field_info) or field_info.repr is False:
+            continue
+        if field_info.is_required():
+            required_field_names.append(name)
+        annotation = resolved_hints.get(name, field_info.annotation)
+        lines.extend(
+            _render_field(name, field_info, annotation=annotation, indent=indent, base_cls=base_cls, depth=depth)
+        )
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer-config/src/data_designer/config/base_utils.py]: Not applicable — discriminator fields are always `Literal['llm-text']` with a matching default. Pydantic resolves these identically to `get_type_hints()`. No config in this codebase uses a type alias for a discriminator Literal.

File: packages/data-designer-config/src/data_designer/config/base_utils.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        return []
+
+    if origin is Union or origin is types.UnionType:
+        non_none = [a for a in args if a is not type(None)]
+        if len(non_none) == 1:
+            return _find_expandable_types(non_none[0], base_cls)
+        return []
+
+    if origin in (list, set, frozenset, tuple):
+        return [t for arg in args for t in _find_expandable_types(arg, base_cls)]
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer-config/src/data_designer/config/base_utils.py]: Stale — tuple handling was removed. `_find_expandable_leaves` now only handles `list` and `Optional`, which are the only patterns used by config fields.

File: packages/data-designer/src/data_designer/cli/main.py [hunk 5]
```diff
@@ -3,8 +3,14 @@
 
 from __future__ import annotations
 
+import warnings
+
 import typer
 
+warnings.filterwarnings("ignore", category=DeprecationWarning, module=r"pyarrow(?:\..*)?")
+warnings.filterwarnings("ignore", category=FutureWarning, module=r"pyarrow(?:\..*)?")
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer/src/data_designer/cli/main.py]: Good catch — moved the filters into `main()` so they only apply during CLI invocation, not at import time.

File: packages/data-designer-config/src/data_designer/config/base_utils.py [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+        return []
+
+    if origin is Union or origin is types.UnionType:
+        non_none = [a for a in args if a is not type(None)]
+        return _find_expandable_leaves(non_none[0], base_cls) if len(non_none) == 1 else []
+
+    if origin is list:
+        return _find_expandable_leaves(args[0], base_cls)
+
+    return []
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer-config/src/data_designer/config/base_utils.py]: Acknowledged — minor style point with no runtime impact as noted. Keeping depth logic in the caller to avoid coupling `_find_expandable_leaves` to rendering concerns.

File: packages/data-designer-config/src/data_designer/config/base_utils.py [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+    """Return True for single-value Literal fields whose default matches the Literal arg."""
+    ann = _unwrap_annotated(info.annotation)
+    if get_origin(ann) is not Literal:
+        return False
+    args = get_args(ann)
+    if len(args) != 1:
+        return False
+    arg = args[0].value if isinstance(args[0], Enum) else args[0]
+    default = info.default.value if isinstance(info.default, Enum) else info.default
+    return arg == default
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer-config/src/data_designer/config/base_utils.py]: The title is misleading — as the analysis confirms, no crash occurs. Discriminator fields always have defaults in this codebase. The existing code correctly returns `False` for required fields.

File: packages/data-designer-config/src/data_designer/config/base_utils.py [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+        # On Python 3.10, generic args can be lost (e.g. list[Score] → bare list);
+        # _recover_annotation falls back to the MRO only for those fields.
+        ann = info.annotation
+        ann_expand = _recover_annotation(cls, name, ann) if _has_degraded_generics(ann) else ann
+
+        if info.is_required():
+            required.append(name)
+            lines.append(f"{indent}  {name}: {_format_type(ann)}  [required]")
+        else:
+            lines.append(f"{indent}  {name}: {_format_type(ann)} = {_format_default(info)}")
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer-config/src/data_designer/config/base_utils.py]: Good catch — fixed. Now using a single `ann` variable for both display and expansion, with the recovered annotation when degraded.

File: packages/data-designer-config/src/data_designer/config/base_utils.py [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+
+        if info.description:
+            lines.append(f"{indent}      {info.description}")
+
+        for leaf in _find_expandable_leaves(ann, base_cls):
+            if issubclass(leaf, Enum):
+                lines.append(f"{indent}      values: {', '.join(str(m.value) for m in leaf)}")
+            elif issubclass(leaf, base_cls) and depth < 1:
+                lines.append(_render_model(leaf, base_cls=base_cls, depth=depth + 1))
```
[the remark this answers is not in the document]
Comment (johnnygreco) [reply packages/data-designer-config/src/data_designer/config/base_utils.py]: Fixed — added an `expanded: set[type]` tracker in `_render_model` so the same ConfigBase subclass is only expanded once, and added `test_duplicate_nested_type_expanded_once` to cover this case.
